### ISSUE TITLE
Exclude not working for AllCops in .rubocop.yml?

### ISSUE THREAD

**Issue by RST-J** [bbatsov/rubocop#2492]
I have some code that I don't want to be checked by the cops and hence configured the respective files to be excluded in `.rubocop.yml`:
```YAML
AllCops:
  Exclude:
    - 'app/controllers/commontator/*.rb'
```
Unfortunately, the cops are run for files within that directory. If I explicitly exclude the same pattern for one of the violated cops, the violation is not reported anymore:
```YAML
Metrics/ClassLength:
  Exclude:
    - 'app/controllers/commontator/*.rb'
```
Any idea why that is the case? I'm using rubocop 0.35.1 and this is the complete config (without the pattern in question):
```YAML
AllCops:
  Exclude:
    - 'db/schema.rb'
    - 'vendor/**/*'
  DisplayCopNames: true
  RunRailsCops: true

Documentation:
  Enabled: false

Metrics/LineLength:
  Max: 180

Metrics/MethodLength:
  Max: 20

```

**Comment by jonas054** [comment]
So if you add the exclusion in question, so that your `.rubocop.yml` becomes
```yaml
AllCops:
  Exclude:
    - 'db/schema.rb'
    - 'vendor/**/*'
    - 'app/controllers/commontator/*.rb'
  DisplayCopNames: true
  RunRailsCops: true

Documentation:
  Enabled: false

Metrics/LineLength:
  Max: 180

Metrics/MethodLength:
  Max: 20
```
and run `rubocop --list`, then files under `app/controllers/commontator/` show up in the list?

**Comment by RST-J** [comment]
No, they don't show up. I think I know why this happens. I have a [rubocop plugin](https://github.com/ngmy/vim-rubocop) for vim which runs rubocop when a file is saved but this does not care about the config rubocop uses when run from the command line. I'll try to figure it out or ask in the plugin repository. Nevermind.

**Comment by jonas054** [comment]
There's an option called `--force-exclusion`, which editor plugins should use. The problem is that when you give a single file name to `rubocop`, then `Exclude` parameters don't come into play. Unless you run with `--force-exclusion`. I see that information about this is missing on the README page.

**Comment by chronodm** [comment]
I was seeing this in 0.46 after not seeing it in 0.43. Turned out I had two `AllCops` stanzas in my `.rubocop.yml`. Once I merged them, the problem went away.

**Comment by jaredbeck** [comment]
> There's an option called `--force-exclusion`, which editor plugins should use. The problem is that when you give a single file name to `rubocop`, then `Exclude` parameters don't come into play. Unless you run with `--force-exclusion`. I see that information about this is missing on the README page.

Can this also be accomplished by [using `--stdin` correctly](https://github.com/rubocop-hq/rubocop/pull/6607), or am I conflating two different things?
